### PR TITLE
[82726] Add E2E tests for the HF tab

### PR DESCRIPTION
**Pull request by akashchi**
### All Submissions:
* [x] Create a meaningful name of the PR with reference to JIRA.
* [x] JIRA tickets:
  * 82726
* [x] Set assignee to the PR.
* [x] Add corresponding labels to the PR. 
* [x] Set appropriate milestone to the PR.
* [x] Added tests for new functionality (unit / e2e)
* [x] Followed code style guidelines:
  * Code covered with typings
  * No debug printing in production code
  * No commented blocks of code in production code

### CODE REVIEW

File: client/e2e/src/pages/hugging-face-model-download.po.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import { browser, by, element, ElementArrayFinder, ElementFinder, protractor } from 'protractor';
+
+import { TestUtils } from './test-utils';
+import { ModelFile } from './model-file';
+import { OMZModelPrecisionEnum } from '../../../src/app/modules/model-manager/components/model-downloader-table/model-downloader-table.component';
+
+export const filterGroupNames = ['task', 'library', 'model-type', 'language', 'license'];
+type FilterGroupName = typeof filterGroupNames[number];
```
Comment (SalnikovIgor) [inline client/e2e/src/pages/hugging-face-model-download.po.ts:8]: As far as I understand, `FilterGroupName` will be a `string` type anyway. If we want to limit to only types from an array `filterGroupNames` , then we can do this:
`type FilterGroupName = 'task' | 'library' | 'model-type' | 'language' | 'license'`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: client/e2e/src/import-hugging-face-models.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  it('select and deselect several filters, check that they are applied correctly', async () => {
+    await testUtils.modelManagerPage.goToModelManager();
+    await testUtils.HFModelDownloadPage.openHFTab();
+
+    // Check that there are several filters available for groups
+    for (const filterGroup of filterGroupNames) {
+      expect(await testUtils.HFModelDownloadPage.countFiltersByGroup(filterGroup)).toBeTruthy();
+    }
+
+    await testUtils.HFModelDownloadPage.selectFilter('bert');
```
[the remark this answers is not in the document]
Comment (akashchi) [reply client/e2e/src/import-hugging-face-models.e2e-spec.ts:76]: There are too many of them to make an exhausting `enum`. Do you suggest creating an incomplete one, just with several values?

File: client/e2e/src/pages/hugging-face-model-download.po.ts (hunk 1 above)
Comment (akashchi) [reply client/e2e/src/pages/hugging-face-model-download.po.ts:8]: Yes, this was a previous iteration. Yet with this approach we would not be able to iterate over `type FilterGroupName`.

File: client/e2e/src/pages/model-download.po.ts [hunk 3]
```diff
@@ -16,34 +16,46 @@ export class ModelDownloadPage {
     modelDescription: TestUtils.getElementByDataTestId('model-description'),
     modelLicense: TestUtils.getElementByDataTestId('model-license'),
     downloadButton: TestUtils.getElementByDataTestId('download-and-import'),
+    precisionContainer: element(by.id('dataType')),
+    convertButton: element(by.buttonText('Convert')),
```
Comment (akashchi) [inline client/e2e/src/pages/model-download.po.ts:20]: This will be addressed in a different PR.
